src/Microsoft.SqlTools.ServiceLayer/AzureFunctions/InsertSqlBindingOperation.cs

```diff
@@ -0,0 +1,143 @@
+                {
+                    return new ResultStatus()
+                    {
+                        Success = false,
+                        ErrorMessage = $"Couldn't find Azure function with FunctionName {Parameters.functionName} in {Parameters.filePath}"
+                    };
+                }
+                else if (azureFunctionMethods.Count() > 1)
+                {
+                    return new ResultStatus()
+                    {
+                        Success = false,
+                        ErrorMessage = $"More than one Azure function found with the FunctionName {Parameters.functionName} in {Parameters.filePath}"
+                    };
+                }
+
+                MethodDeclarationSyntax azureFunction = azureFunctionMethods.First();
+                var newParam = this.Parameters.bindingType == BindingType.input ? this.GenerateInputBinding() : this.GenerateOutputBinding();
+
+                // Generate updated method with the new parameter
+                // normalizewhitespace gets rid of any newline whitespace in the leading trivia, so we add that back
+                var updatedMethod = azureFunction.AddParameterListParameters(newParam).NormalizeWhitespace().WithLeadingTrivia(azureFunction.GetLeadingTrivia()).WithTrailingTrivia(azureFunction.GetTrailingTrivia());
+
+                // Replace the node in the tree
+                root = root.ReplaceNode(azureFunction, updatedMethod);
+
+                // write updated tree to file
+                var workspace = new AdhocWorkspace();
+                var syntaxTree = CSharpSyntaxTree.ParseText(root.ToString());
+                var formattedNode = CodeAnalysis.Formatting.Formatter.Format(syntaxTree.GetRoot(), workspace);
+                StringBuilder sb = new StringBuilder(formattedNode.ToString());
+                string content = sb.ToString();
+                File.WriteAllText(Parameters.filePath, content);
+
+                return new ResultStatus()
+                {
+                    Success = true
+                };
+            }
+            catch (Exception e)
+            {
+                return new ResultStatus()
+                {
+                    Success = false,
+                    ErrorMessage = e.ToString()
+                };
+            }
+        }
+
+        /// <summary>
+        /// Generates a parameter for the sql input binding that looks like
+        /// [Sql("select * from [dbo].[table1]", CommandType = System.Data.CommandType.Text, ConnectionStringSetting = "SqlConnectionString")] IEnumerable<Object> result
+        /// </summary>
+        private ParameterSyntax GenerateInputBinding()
+        {
+            // Create arguments for the Sql Input Binding attribute
+            var argumentList = SyntaxFactory.AttributeArgumentList();
+            argumentList = argumentList.AddArguments(SyntaxFactory.AttributeArgument(SyntaxFactory.IdentifierName($"\"select * from {Parameters.objectName}\"")));
```

Comment:
to make this really useful, we should support passing in query text as well, right?

Reply:
For this initial implementation, I think we discussed keeping this pretty simple, just so that all the boiler plate bindings stuff gets added to the Azure function, then the user can update the query later. Definitely something we should think more about doing later, but in a previous discussion, it was brought up that it should have a more rich support for writing the query than just typing in a quick pick.